kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -761,21 +779,68 @@ class ResidualBasedNewtonRaphsonStrategy
         //initializing the parameters of the Newton-Raphson cycle
         unsigned int iteration_number = 1;
         r_model_part.GetProcessInfo()[NL_ITERATION_NUMBER] = iteration_number;
-        bool is_converged = false;
+        //bool is_converged = false;
         bool residual_is_updated = false;
         p_scheme->InitializeNonLinIteration(r_model_part, rA, rDx, rb);
         mpConvergenceCriteria->InitializeNonLinearIteration(r_model_part, r_dof_set, rA, rDx, rb);
-        is_converged = mpConvergenceCriteria->PreCriteria(r_model_part, r_dof_set, rA, rDx, rb);
+        bool is_converged = mpConvergenceCriteria->PreCriteria(r_model_part, r_dof_set, rA, rDx, rb);
 
         // Function to perform the building and the solving phase.
         if (BaseType::mRebuildLevel > 0 || BaseType::mStiffnessMatrixIsBuilt == false) {
             TSparseSpace::SetToZero(rA);
             TSparseSpace::SetToZero(rDx);
             TSparseSpace::SetToZero(rb);
 
-            p_builder_and_solver->BuildAndSolve(p_scheme, r_model_part, rA, rDx, rb);
+            if (mUseOldStiffnessInFirstIteration)
+            {
+                KRATOS_INFO_IF("NR-Strategy", this->GetEchoLevel() > 0)
```

Comment:
A small comment, not very critical but can help. 
It would be nice to take the implementation in this `if` condition to a new function. For example a private function. 
With these changes function `SolveSolutionStep` will be close to 200 line long.

Reply:
i completely agree. I will do once i found the "definitive" solution.